add validation checker

evakalo opened the pull request:
added email validation

evakalo committed:
add validation checker

tonkec commented on src/components/auth/IsEmailValid.js:
```diff
@@ -0,0 +1,5 @@
+export default function ValidateEmail(email) {
+  let emailFormat =
```
why are you using let instead of const

tonkec commented on src/components/auth/Login/index.js:
```diff
@@ -1,33 +1,44 @@
+import { useNavigate, Link } from "react-router-dom";
+import AuthLayout from "../../Layout/AuthLayout";
+import "./../Auth.scss";
+import IsEmailValid from "../IsEmailValid";
```
only components name are uppercased

tonkec commented:
- please if email is not valid, button should be disabled
- create a reusable <Error /> component
- add a single css component that will hold styles for error messages

tonkec commented on src/components/auth/Login/index.js:
```diff
@@ -1,33 +1,44 @@
     return state.authReducer.isVerified;
   });
 
+  const setAndCheckEmail = (e) => {
```
add two functions instead of one

evakalo committed:
error component, button disable

added error compnent, button disabled when email and password are in an incorrect format.
problem with inputs when editing

tonkec committed:
add test env to gitignore

tonkec committed:
- refactor architecture
  - move validators to their own file
  - add two methods for handling valid and invalid input
  - move error messages to constants
  - move Error component to components
  - disable form submission if there is an error even if the button is enabled
